.github/workflows/linux_builds.yml

```diff
@@ -212,3 +213,57 @@ jobs:
         if: ${{ matrix.artifact }}
         with:
           name: ${{ matrix.cache-name }}
+
+      - name: Dump GDExtension API for compatibility checking
+        if: ${{ matrix.compat }}
+        run: |
+          ${{ matrix.bin }} --headless --dump-extension-api
+
+      - name: Upload extension API
+        uses: actions/upload-artifact@v3
+        if: ${{ matrix.compat }}
+        with:
+          name: extension-api
+          path: extension_api.json
+          # Use the default artifact retention of 90 days to ensure the compat checking
+          # step below works as long as possible for PRs based on old commits
+
+      # Last step of this workflow as when merging a PR that intentionally breaks compat
+      # the failure of this step must not hide any other issues
+      - name: Check for GDExtension compatibility
+        if: ${{ matrix.compat }}
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          # download the reference extension_api.json from the previous build that compatibility should be checked against
+          get_api_ref() {
+            # fetch the git history so that we can get the previous commits
+            # but exclude diffs etc to keep the size down
+            git fetch --unshallow --filter=tree:0 > /dev/null 2> /dev/null
+            # For PRs check compatibility against the base commit, ie check the compatibility breaks from the entire PR if it has multiple commits
+            # For commits on branches check for compatibility against the previous commit
+            parent="${{ github.event.pull_request.base.sha }}"
+            if [ -z "$parent" ]; then parent="${{ github.sha }}^"; fi
+            # For the parent commit (and up to 100 parents, in case of batch merges or if the CI of some commits hasn't completed yet)
+            while read -r commit; do
+              # For each workflow id for the commit
+              while read -r workflow_id; do
+                # For each artifact named 'extension-api' of the workflow
```

Comment:
Instead of having to add two extra stages above to create and upload an `extension-api` artifact, would it be possible to just download the `linux-editor-mono` binary here, then dump and validate the extension apis in a single run below?

That would make this job much more reusable as a foundation for other types of regression tests, such as @Calinou 's https://github.com/Calinou/godot-rendering-tests

> The step failing on this PR is expected, as no api to compare to exists before this PR, similarly the merge commit for this PR will also fail.
> 
This (admittedly minor) issue would also be resolved.